### DISCUSSION TITLE
OpenGraph (metadata)  in Next Internationalization

### DISCUSSION THREAD

**Discussion by Minhnhat0408** [vercel/next.js#55497] [Help] [answered]
### Summary

so I set up my app according to [Internationalization (i18n) Routing](https://nextjs.org/docs/pages/building-your-application/routing/internationalization) this is my folder structure:

<img width="244" alt="image" src="https://github.com/vercel/next.js/assets/63915841/85924a0a-ea87-4c7d-9d78-7f6114357a86">

Folder structure

It work completely fine. However it only have the metadata for the route '[lang]' ('https//examples.com/vi') or ('https//examples.com/en') but with the ('https//examples.com/') it doesn't show anything. Even if I make a layout.tsx in the 'app' directly with the metadata in it. Is there anyway to have metadata even in the ('https//examples.com/').
Note: My middleware is redirect the ('https//examples.com/') to an lang route not sure whether it the problem.

### Additional information

You can find my code here [Code](https://github.com/Minhnhat0408/Hub-Media)

### Example

https://pbhubmedia.vercel.app/

**Comment by suskarkhy** [comment]
I have the exact same issue. were you able to fix it?

**Reply by Minhnhat0408** [reply]
Unfortunately no. I try to found any possible solutions but to no avail

**Comment by denis23x** [comment]
Hi @Minhnhat0408 

In this guide [Internationalization (i18n) Routing](https://nextjs.org/docs/pages/building-your-application/routing/internationalization) you can find the next:

> Automatic Locale Detection
>
> When a user visits the application root (generally `/`), Next.js will try to automatically detect which locale the user prefers based on the `Accept-Language` header and the current domain.
>
> If a locale other than the default locale is detected, the user will be redirected to either:
>
> - **When using Sub-path Routing**: The locale prefixed path
> - **When using Domain Routing**: The domain with that locale specified as the default
>
> When using Domain Routing, if a user with the `Accept-Language` header `fr;q=0.9` visits `example.com`, they will be redirected to `example.fr` since that domain handles the `fr` locale by default.
> 
> When using Sub-path Routing, the user would be redirected to `/fr`.


So I guess your project working fine..

If I misunderstood something please clarify it for me 🙏

**Comment by Minhnhat0408** [comment]
Hi @denis23x 

Yes it automatically detect the locale and redirect user to the right locale path. However the problem is that the opengraph of the page dont exist if you only use [https://pbhubmedia.vercel.app](https://pbhubmedia.vercel.app) , it only exists if you have specify the locale path like [https://pbhubmedia.vercel.app/en](https://pbhubmedia.vercel.app/en) or [https://pbhubmedia.vercel.app/vi](https://pbhubmedia.vercel.app/vi)

This is a real problem if you send the link to someone that you dont know their language, the link will have no opengraph and it look like a not verified website link 

![image](https://github.com/user-attachments/assets/68e39150-828f-4484-b386-cc98341b97c8)

**Reply by denis23x** [reply, marked as answer]
Ahh, now I got you 👌

Well, look, the thing is that when you share a link, the metainformation is received by the server of the service in which you share the link, and not by the user client.

Therefore, it can be assumed that the bot that runs after meta information does not have the required header `Accept-Language` in the request and automatic locale detection just not working.

If you want automatic redirect anyway (to default language, because there is no way you can detect right user language from the crawling bot), you will have to write middleware for that.

You can find an example in the documentation in the link above.
